extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -216,6 +221,18 @@ public ServiceStartBuildItem startActions(FlywayRecorder recorder,
         return new ServiceStartBuildItem("flyway");
     }
 
+    @BuildStep
+    void configureKubernetes(ContainerImageInfoBuildItem imageInfo,
```

Comment:
is there a reason why flyway will have to be dependent on kubernetes for this? 

wouldn't this be possible to be generic and have notion of running init tasks vs running fully?

Reply:
All extensions that influence the generation of kubernetes manifests need to depend on the kubernetes spi.
The kubernetes spi is a zero dependency (I think) module that only defines build items realted to the generation of the manfiests.

There is no technical reason for not using something more abstract other than that we don't have something in place.
We could possibly define an abstract Job or Init Task and have it translate to kubernetes, amazon lambda, or maybe in the future tekton etc. I can give it a spin and see how it goes.